docs/api/v3.rst

```diff
@@ -0,0 +1,727 @@
+    :>json string built: Whether this version has been built
+    :>json string active: Whether this version is active
+    :>json string type: The type of this version (typically "tag" or "branch")
+    :>json string last_build: Build object representing the last build of this version
+    :>json array downloads: URLs to downloads of this version's documentation
+
+    :requestheader Authorization: optional token to authenticate.
+
+    :statuscode 200: Success
+    :statuscode 404: There is no ``Version`` with this slug for this project
+
+
+Version edit
+++++++++++++
+
+.. http:patch:: /api/v3/projects/(string:project_slug)/version/(string:version_slug)/
+
+    Edit a version.
+
+    **Example request**:
+
+    .. sourcecode:: json
+
+        {
+            "active": true,
+            "privacy_level": "public"
+        }
+
+    **Example response**:
+
+    `See Version details <#version-detail>`_
+
+    :requestheader Authorization: required token to authenticate.
+
+    :statuscode 204: Edited sucessfully
+    :statuscode 400: Some field is invalid
+    :statuscode 401: Not valid permissions
+    :statuscode 404: There is no ``Version`` with this slug for this project
+
+
+Builds
+~~~~~~
+
+Builds are created by Read the Docs whenever a ``Project`` has its documentation built.
+Frequently this happens automatically via a web hook but can be triggered manually.
+
+Builds can be viewed in the build page for a project.
+For example, here is `Pip's build page`_.
+
+.. _Pip's build page: https://readthedocs.org/projects/pip/builds/
+
+
+Build details
++++++++++++++
+
+.. http:get:: /api/v3/projects/(str:project_slug)/builds/(int:build_id)/
+
+    Retrieve details of a single build for a project.
+
+    **Example request**:
+
+    .. sourcecode:: bash
+
+        $ curl https://readthedocs.org/api/v3/projects/pip/builds/8592686/?include_config=true
+
+    **Example response**:
+
+    .. sourcecode:: json
+
+        {
+            "id": 8592686,
+            "version": "latest",
+            "project": "pip",
+            "created": "2018-06-19T15:15:59+00:00",
+            "finished": "2018-06-19T15:16:58+00:00",
+            "duration": 59,
+            "state": {
+                "code": "finished",
+                "name": "Finished"
+            },
+            "success": true,
+            "error": null,
+            "commit": "6f808d743fd6f6907ad3e2e969c88a549e76db30",
+            "builder": "build03",
+            "cold_storage": false,
+            "config": {
+                "version": "1",
+                "formats": [
+                    "htmlzip",
+                    "epub",
+                    "pdf"
+                ],
+                "python": {
+                    "version": 3,
+                    "install": [
+                        {
+                            "requirements": ".../stable/tools/docs-requirements.txt"
```

Comment:
Hmm, I want to change this to be relative paths, doesn't look like info the user cares about (the whole path in the servers)

Reply:
This is another issue btw

Reply:
This comes from the db as is. I didn't manipulate it.

I'm not sure how to handle this yet, if we will need to create a Serializer for the config and define all of the fields with a fixed structure or just rely on the `config` object from the db.

Reply:
Yeah, I mean, I should create another issue for this. Have been bothering me for a while